File: packages/pensopay_android/android/src/main/kotlin/com/flutter/pensopay/networking/pensopayapi/pensopaylink/acquirers/PPGetAcquireSettingsClearhausRequest.kt
package com.flutter.pensopay.networking.pensopayapi.pensopaylink.acquirers

import com.android.volley.Request
import com.flutter.pensopay.networking.pensopayapi.PPrequest
import org.json.JSONObject

internal class PPGetAcquireSettingsClearhausRequest(): PPrequest<PPClearhausSettings>(Request.Method.GET, "/acquirers/clearhaus", null, PPClearhausSettings::class.java) {}

internal class PPClearhausSettings : JSONObject() {

    var active: Boolean = false
    var api_key: String = ""
    var apple_pay: Boolean = false
    var recurring: Boolean = false
    var payout: Boolean = false
    var mpi_merchant_id: String? = null

}

File: packages/pensopay_ios/ios/Classes/Networking/Pensopay API/Pensopay Link/Acquirers/PPGetAcquireSettingsClearhausRequest.swift
//
//  PPGetAcquireSettingsClearhausRequest.swift
//  PensopaySDK
//
//  Created on 07/26/2022
//  Copyright © 2022 Pensopay. All rights reserved.
//

import Foundation

public class PPGetAcquireSettingsClearhausRequest: PPRequest {
    
    // MARK: - Init
    
    public override init() {
        super.init()
    }
    
    
    // MARK: - URL Request
    
    public func sendRequest(success: @escaping (_ result: PPClearhausSettings) -> Void, failure: ((_ data: Data?, _ response: URLResponse?, _ error: Error?) -> Void)?) {
        guard let url = URL(string: "\(PensopayAPIBaseUrl)/acquirers/clearhaus") else {
            return
        }
        
        var request = URLRequest(url: url)
        request.httpMethod = "GET"
        request.setValue(headers.encodedAuthorization(), forHTTPHeaderField: "Authorization")
        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
        request.setValue("application/json", forHTTPHeaderField: "Accept")
        
        super.sendRequest(request: request, success: success) { (data, response, error) in
            if let httpUrlResponse = response as? HTTPURLResponse {
                if httpUrlResponse.statusCode == PensopayHttpStatusCodes.unauthorized.rawValue {
                    Pensopay.logDelegate?.log("The API key needs permissions for 'GET  /acquirers/clearhaus'")
                }
            }
            
            failure?(data, response, error)
        }
    }
    
}

public class PPClearhausSettings: Codable {
    
    public var active: Bool
    public var api_key: String
    public var apple_pay: Bool
    public var recurring: Bool
    public var payout: Bool
    public var mpi_merchant_id: String?
    
}
